spa/src/api/rest-api.ts

```diff
@@ -0,0 +1,75 @@
+export function createRestApi() {
+
+	let gitHubAccessToken: string | undefined;
+	let gitHubRefreshToken: string | undefined;
+
+	return {
+		storeGitHubToken,
+		isTokenStillValid,
+		generateOAuthUrl,
+		fetchInstalledOrganisations
+	};
+
+	function storeGitHubToken(tokens: { gitHubAccessToken: string; githubRefreshToken?: string }) {
+		gitHubAccessToken = tokens.gitHubAccessToken;
+		gitHubRefreshToken = tokens.githubRefreshToken;
+	}
+
+	async function isTokenStillValid() {
+
+		try {
+			const resp = await fetch("https://api.github.com/user", {
+				headers: {
+					Authorization: `Bearer ${gitHubAccessToken}`
+				}
+			});
+			if (!resp.ok) {
```

Comment:
This is a catch of `fetch`, not sure whether you awared of it, `fetch` doesn't throw on 400/500, it only throw on network problem. Need to check `resp.ok` to determine whether it is 200